src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -253,11 +297,77 @@ def extract_archive_argv(self, archive_path: str, extract_path: str) -> tuple[st
         return (self.path, "xf", archive_path, "-C", extract_path)
 
 
+class MkdirBinary(BinaryPath):
+    pass
+
+
+class ChmodBinary(BinaryPath):
+    pass
+
+
 # -------------------------------------------------------------------------------------------
-# Rules to find binaries
+# Binaries Rules
 # -------------------------------------------------------------------------------------------
 
 
+@rule
+async def create_binary_shims(
+    binary_shims_request: BinaryShimsRequest,
+    bash: BashBinary,
+    mkdir: MkdirBinary,
+    chmod: ChmodBinary,
+) -> BinaryShims:
+    """Creates a bin directory with shims for all requested binaries.
+
+    Useful as input digest for a Process to setup a `bin` directory for PATH.
+    """
+    requests = binary_shims_request.requests
+    all_binary_paths = await MultiGet(
+        Get(BinaryPaths, BinaryPathRequest, request) for request in requests
+    )
```

Comment:
Ah, after looking at this code being used in https://github.com/pantsbuild/pants/pull/14361, I'm wondering what you think about callers having to call this beforehand and pass in the `BinaryPaths` results (or probably the chosen path)

Why? We want to encourage using the pre-built binary tips like `UnzipBinary` and `TarBinary`. I want to make sure that you can use those with this new shim rule you wrote.

Wdyt?

Reply:
I'll take a look at it. I like it being wrapped like this, but that's only useful when the things you want is specified elsewhere, which probably is rarer than the normal case.

Reply:
@Eric-Arellano Suggested fix adding support for passing `BinaryPath` objects rather than requests in #14789